### ISSUE TITLE
guide

### ISSUE THREAD

**Issue by thisisAbox** [HiromuHota/pentaho-kettle#209]
Could you please give a guidence document about how to run this project in IntelJ Idea?

**Comment by HiromuHota** [comment]
Because of Karaf, it is not easy to run webSpoon on IDE (Eclipse, IntelliJ, etc.).
Instead, I'd recommend to build&deploy as a Docker container and remotely debug it.
https://github.com/HiromuHota/pentaho-kettle/tree/webspoon-9.0/docker#debug

**Comment by thisisAbox** [comment]
I think debug it remotely is difficult to develop this project?
I was wondered how you guys to develop this project? Don't you use IDE? 
Now i add a web module in this project and i run it successfully, but  there are some problems when i click the connect button,the New Repository Connection Interface is blank, i can't handle it! please give a method for me!

**Comment by HiromuHota** [comment]
I use Eclipse for coding and remote debugging. I think you can use IntelliJ for these too.
(See here https://www.jetbrains.com/help/idea/tutorial-remote-debug.html#98293 for how to remotely debug).

It is possible to run webSpoon on IDE like Eclipse and IntelliJ as you demonstrated.
The problem is that webSpoon requires the `system` folder at the current directory, which includes Karaf.
The reason why the repository connect button does not work would be that you did not place the folder at the current directory.
In other words, if you place the `system` folder at the current directory, you can run it on IDE.
The next question you would have is that where is the "current directory"?
It depends how your IDE run it. You would need to figure it out yourself.

As I find it cumbersome and error-prone, I build&deploy as a Docker container and remotely debug it using Eclipse.
If you can find an easy solution, it'd be great! Please let me know!
